src/lightning_lite/plugins/precision/precision.py

```diff
@@ -0,0 +1,96 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import contextlib
+from typing import Any, Dict, Generator, Optional, Union
+
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from lightning_lite.utilities.types import _PARAMETERS
+
+
+class PrecisionPlugin:
+    """Base class for all plugins handling the precision-specific parts of the training.
+
+    The class attribute precision must be overwritten in child classes. The default value reflects fp32 training.
+    """
+
+    precision: Union[str, int] = 32
+
+    def get_main_params(self, optimizer: Optimizer) -> _PARAMETERS:
+        """The main params of the model.
+
+        Returns the plain model params here. Maybe different in other precision plugins.
+        """
+        for group in optimizer.param_groups:
+            yield from group["params"]
+
+    @contextlib.contextmanager
+    def forward_context(self) -> Generator[None, None, None]:
+        """A contextmanager for managing model forward/training_step/evaluation_step/predict_step."""
+        yield
+
+    def pre_backward(self, tensor: Tensor, module: Optional[Module]) -> None:
+        """Run before precision plugin executes backward.
+
+        Args:
+            tensor:
+            module:
```

Comment:
Is this empty on purpose? I would fill it out or remove it. Other instances below